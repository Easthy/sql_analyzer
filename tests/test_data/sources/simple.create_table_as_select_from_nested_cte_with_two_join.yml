source.users:
  - id
  - name
  - project_id
source.user_status:
  - user_id
  - status
reference.project:
  - project_id
  - name