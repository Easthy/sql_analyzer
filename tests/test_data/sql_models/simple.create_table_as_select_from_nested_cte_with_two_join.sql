CREATE TABLE simple.simple.create_table_as_select_from_nested_cte_with_two_join AS
WITH _cte AS (
      WITH _nested_cte AS (
              SELECT users.id AS user_id,
                     users.name,
                     project.project_id
                FROM source.users

                     INNER JOIN reference.project
                     ON users.project_id = users.project_id
      )
      SELECT _nested_cte.*,
             user_status.status
        FROM _nested_cte

             LEFT JOIN source.user_status
             ON user_status.user_id = _nested_cte.user_id
)
SELECT _cte.user_id,
       _cte.name,
       _cte.status,
       _cte.project_id
  FROM _cte
; 
