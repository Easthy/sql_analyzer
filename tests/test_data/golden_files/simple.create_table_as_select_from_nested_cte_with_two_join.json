{
  "nodes": [
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join.name" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join.user_id" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join.status" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join.project_id" },
    { "id": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join" },
    { "id": "col:source.users.name" },
    { "id": "col:source.users.id" },
    { "id": "col:source.users.project_id" },
    { "id": "tbl:source.users" },
    { "id": "col:source.user_status.user_id" },
    { "id": "col:source.user_status.status" },
    { "id": "tbl:source.user_status" },
    { "id": "tbl:reference.project" },
    { "id": "col:reference.project.project_id" },
    { "id": "col:reference.project.name" }
  ],
  "links": [
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join.name", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join.user_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join.status", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join.project_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "tbl:source.users", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "tbl:source.user_status", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join", "target": "tbl:reference.project", "type": "table_dependency" },
    { "source": "tbl:source.users", "target": "col:source.users.name", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.id", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.project_id", "type": "contains_column" },
    { "source": "tbl:source.user_status", "target": "col:source.user_status.status", "type": "contains_column" },
    { "source": "tbl:source.user_status", "target": "col:source.user_status.user_id", "type": "contains_column" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join.name", "target": "col:source.users.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join.user_id", "target": "col:source.users.id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join.status", "target": "col:source.user_status.status", "type": "column_dependency" },
    { "source": "tbl:reference.project", "target": "col:reference.project.name", "type": "contains_column" },
    { "source": "tbl:reference.project", "target": "col:reference.project.project_id", "type": "contains_column" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join.project_id", "target": "col:reference.project.project_id", "type": "column_dependency" }
  ]
}